lib/plugins/aws/package/compile/functions/index.test.js

```diff
@@ -2831,4 +2810,207 @@ describe('AwsCompileFunctions #2', () => {
+          },
+        },
+      });
+
+      expect(firstVersionId).to.equal(
+        data.cfTemplate.Outputs.FooLambdaFunctionQualifiedArn.Value.Ref
+      );
+    });
+
+    it('should create different lambda version id for different property keys (but no different values)', async () => {
+      const firstVersionId = firstCfTemplate.Outputs.FooBarLambdaFunctionQualifiedArn.Value.Ref;
+
+      await updateConfig({
+        functions: {
+          fooBar: { handler: 'index.handler', provisionedConcurrency: 1 },
+        },
+      });
+
+      const data = await runServerless({
+        cwd: servicePath,
+        cliArgs: ['package'],
+        awsRequestStubMap: {
+          CloudFormation: {
+            describeStacks: { Stacks: [{ Outputs: [{ OutputKey: 'test' }] }] },
+          },
+        },
+      });
+
+      expect(firstVersionId).to.equal(
+        data.cfTemplate.Outputs.FooBarLambdaFunctionQualifiedArn.Value.Ref
+      );
+    });
+
+    it('should create same version id for same layer zip content', async () => {
+      const firstVersionId = firstCfTemplate.Outputs.FooLambdaFunctionQualifiedArn.Value.Ref;
+      const originalLayer = path.join(servicePath, 'testLayer');
+      const sourceChangeLayer = path.join(servicePath, 'extra_layers', 'testLayerDuplicate');
+      const backupLayer = path.join(servicePath, 'extra_layers', 'testLayerBackup');
+
+      await fse.rename(originalLayer, backupLayer);
+      await fse.rename(sourceChangeLayer, originalLayer);
+
+      const data = await runServerless({
+        cwd: servicePath,
+        cliArgs: ['package'],
+        awsRequestStubMap: {
+          CloudFormation: {
+            describeStacks: { Stacks: [{ Outputs: [{ OutputKey: 'test' }] }] },
+          },
+        },
+      });
+
+      await fse.rename(originalLayer, sourceChangeLayer);
+      await fse.rename(backupLayer, originalLayer);
+
+      expect(firstVersionId).to.equal(
+        data.cfTemplate.Outputs.FooLambdaFunctionQualifiedArn.Value.Ref
+      );
+    });
+
+    it('should create different lambda version id when layer content changed', async () => {
+      const firstVersionId = firstCfTemplate.Outputs.FooLambdaFunctionQualifiedArn.Value.Ref;
+      const originalLayer = path.join(servicePath, 'testLayer');
+      const sourceChangeLayer = path.join(servicePath, 'extra_layers', 'testLayerSourceChange');
+      const backupLayer = path.join(servicePath, 'extra_layers', 'testLayerBackup');
+
+      await fse.rename(originalLayer, backupLayer);
+      await fse.rename(sourceChangeLayer, originalLayer);
+
+      const data = await runServerless({
+        cwd: servicePath,
+        cliArgs: ['package'],
+        awsRequestStubMap: {
+          CloudFormation: {
+            describeStacks: { Stacks: [{ Outputs: [{ OutputKey: 'test' }] }] },
+          },
+        },
+      });
+
+      await fse.rename(originalLayer, sourceChangeLayer);
+      await fse.rename(backupLayer, originalLayer);
```

Comment:
It would be nice to construct this with `before` and `after`, as otherwise if tests fail, the changed state will be leftover, possibly affecting results of other tests

Reply:
Wasn't too sure exactly how to do this as the `before` and `after` need to be run specifically for this test only. I ended up using a nested `describe` with `beforeEach` and `afterEach`.

Reply:
Yes, in that case it'll require nesting extra `describe` block (and imo it's nothing wrong with that)